src/linux/build_id_reader.rs

```diff
@@ -0,0 +1,225 @@
+use crate::errors::BuildIdReaderError as Error;
+use crate::minidump_format::GUID;
+use goblin::{
+    container::{Container, Ctx, Endian},
+    elf,
+};
+
+const NOTE_SECTION_NAME: &[u8] = b".note.gnu.build-id\0";
+
+pub trait ModuleMemory {
```

Comment:
Not sure what the point of using a trait here is if there is only one implementor

Reply:
There are two. https://github.com/rust-minidump/minidump-writer/pull/112/files#diff-ed2349064f78dacbfa376d37df89b784688730cc1a397858b847cc5399f142a9R584-R600